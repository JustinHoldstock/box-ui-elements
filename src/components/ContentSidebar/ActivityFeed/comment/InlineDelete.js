/**
 * @flow
 * @file Inline Delete component
 */

import React, { Component, ReactNode } from 'react';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
import classNames from 'classnames';

import PlainButton from 'box-react-ui/lib/components/plain-button';
import { Flyout, Overlay } from 'box-react-ui/lib/components/flyout';
import IconTrash from 'box-react-ui/lib/icons/general/IconTrash';

import messages from '../messages';

type Props = {
    onDelete: Function,
    id: string,
    intl: intlShape.isRequired,
    message: ReactNode
};

class InlineDelete extends Component<Props> {
    state = {
        isConfirming: false
    };

    onDeleteConfirmedHandler = () => {
        const { id, onDelete } = this.props;
        onDelete({ id });
    };

    handleFlyoutOpen = () => {
        this.setState({ isConfirming: true });
    };

    handleFlyoutClose = () => {
        this.setState({ isConfirming: false });
    };

    render() {
        const { intl, message } = this.props;
        return (
            <div
                className={classNames('box-ui-comment-delete-container', {
                    'is-confirming': this.state.isConfirming
                })}
            >
                <Flyout onClose={this.handleFlyoutClose} onOpen={this.handleFlyoutOpen} position='middle-left'>
                    <PlainButton
                        aria-label={intl.formatMessage(messages.deleteLabel)}
                        className='box-ui-comment-delete'
                    >
                        <IconTrash />
                    </PlainButton>
                    <Overlay>
                        <b>{message}</b>
                        <div>
                            <PlainButton
                                className='lnk box-ui-comment-delete-yes'
                                onClick={this.onDeleteConfirmedHandler}
                                type='button'
                            >
                                <FormattedMessage {...messages.commentDeleteConfirm} />
                            </PlainButton>
                            {' / '}
                            <PlainButton className='lnk box-ui-comment-delete-no' type='button'>
                                <FormattedMessage {...messages.commentDeleteCancel} />
                            </PlainButton>
                        </div>
                    </Overlay>
                </Flyout>
            </div>
        );
    }
}

export { InlineDelete as InlineDeleteBase };
export default injectIntl(InlineDelete);
