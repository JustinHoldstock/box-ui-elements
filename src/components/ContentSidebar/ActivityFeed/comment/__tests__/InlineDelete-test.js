import React from 'react';
import { shallow } from 'enzyme';

import { InlineDeleteBase as InlineDelete } from '../InlineDelete';

const intl = { formatMessage: () => {} };
const translationProps = {
    intl
};

describe('features/activity-feed/comment/InlineDelete', () => {
    const render = (props = {}) =>
        shallow(<InlineDelete {...translationProps} id='123' message='test' onDelete={() => {}} {...props} />);

    test('should correctly render comment', () => {
        const wrapper = render();
        expect(wrapper).toMatchSnapshot();
    });

    test('should set is-confirming class when confirm flyout is open', () => {
        const wrapper = render();
        const flyout = wrapper.find('Flyout');
        expect(wrapper.hasClass('is-confirming')).toBe(false);
        flyout.prop('onOpen')();
        wrapper.update();
        expect(wrapper.hasClass('is-confirming')).toBe(true);
        flyout.prop('onClose')();
        wrapper.update();
        expect(wrapper.hasClass('is-confirming')).toBe(false);
    });

    test('should call onDelete handler when comment deletion is confirmed', () => {
        const onDeleteSpy = jest.fn();
        const wrapper = render({ onDelete: onDeleteSpy });

        const yesBtn = wrapper.find('.bcs-comment-delete-yes');
        yesBtn.simulate('click');

        expect(onDeleteSpy).toHaveBeenCalledWith({ id: '123' });
    });
});
