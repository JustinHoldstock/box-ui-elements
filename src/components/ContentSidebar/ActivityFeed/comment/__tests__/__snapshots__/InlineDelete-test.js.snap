// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/activity-feed/comment/InlineDelete should correctly render comment 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InlineDelete
    id="123"
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    message="test"
    onDelete={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Flyout
        className=""
        closeOnClick={true}
        closeOnClickOutside={true}
        constrainToScrollParent={true}
        constrainToWindow={false}
        isVisibleByDefault={false}
        onClose={[Function]}
        onOpen={[Function]}
        position="middle-left"
      >
        <PlainButton
          aria-label={undefined}
          className="bcs-comment-delete"
        >
          <IconTrash />
        </PlainButton>
        <Overlay>
          <b>
            test
          </b>
          <div>
            <PlainButton
              className="lnk bcs-comment-delete-yes"
              onClick={[Function]}
              type="button"
            >
              <FormattedMessage
                defaultMessage="Yes"
                id="boxui.activityFeed.commentDeleteConfirm"
              />
            </PlainButton>
             / 
            <PlainButton
              className="lnk bcs-comment-delete-no"
              type="button"
            >
              <FormattedMessage
                defaultMessage="No"
                id="boxui.activityFeed.commentDeleteCancel"
              />
            </PlainButton>
          </div>
        </Overlay>
      </Flyout>,
      "className": "bcs-comment-delete-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PlainButton
            aria-label={undefined}
            className="bcs-comment-delete"
          >
            <IconTrash />
          </PlainButton>,
          <Overlay>
            <b>
              test
            </b>
            <div>
              <PlainButton
                className="lnk bcs-comment-delete-yes"
                onClick={[Function]}
                type="button"
              >
                <FormattedMessage
                  defaultMessage="Yes"
                  id="boxui.activityFeed.commentDeleteConfirm"
                />
              </PlainButton>
               / 
              <PlainButton
                className="lnk bcs-comment-delete-no"
                type="button"
              >
                <FormattedMessage
                  defaultMessage="No"
                  id="boxui.activityFeed.commentDeleteCancel"
                />
              </PlainButton>
            </div>
          </Overlay>,
        ],
        "className": "",
        "closeOnClick": true,
        "closeOnClickOutside": true,
        "constrainToScrollParent": true,
        "constrainToWindow": false,
        "isVisibleByDefault": false,
        "onClose": [Function],
        "onOpen": [Function],
        "position": "middle-left",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "aria-label": undefined,
            "children": <IconTrash />,
            "className": "bcs-comment-delete",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <b>
                test
              </b>,
              <div>
                <PlainButton
                  className="lnk bcs-comment-delete-yes"
                  onClick={[Function]}
                  type="button"
                >
                  <FormattedMessage
                    defaultMessage="Yes"
                    id="boxui.activityFeed.commentDeleteConfirm"
                  />
                </PlainButton>
                 / 
                <PlainButton
                  className="lnk bcs-comment-delete-no"
                  type="button"
                >
                  <FormattedMessage
                    defaultMessage="No"
                    id="boxui.activityFeed.commentDeleteCancel"
                  />
                </PlainButton>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": "b",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <PlainButton
                    className="lnk bcs-comment-delete-yes"
                    onClick={[Function]}
                    type="button"
                  >
                    <FormattedMessage
                      defaultMessage="Yes"
                      id="boxui.activityFeed.commentDeleteConfirm"
                    />
                  </PlainButton>,
                  " / ",
                  <PlainButton
                    className="lnk bcs-comment-delete-no"
                    type="button"
                  >
                    <FormattedMessage
                      defaultMessage="No"
                      id="boxui.activityFeed.commentDeleteCancel"
                    />
                  </PlainButton>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <FormattedMessage
                      defaultMessage="Yes"
                      id="boxui.activityFeed.commentDeleteConfirm"
                    />,
                    "className": "lnk bcs-comment-delete-yes",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "defaultMessage": "Yes",
                      "id": "boxui.activityFeed.commentDeleteConfirm",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                " / ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <FormattedMessage
                      defaultMessage="No"
                      id="boxui.activityFeed.commentDeleteCancel"
                    />,
                    "className": "lnk bcs-comment-delete-no",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "defaultMessage": "No",
                      "id": "boxui.activityFeed.commentDeleteCancel",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Flyout
          className=""
          closeOnClick={true}
          closeOnClickOutside={true}
          constrainToScrollParent={true}
          constrainToWindow={false}
          isVisibleByDefault={false}
          onClose={[Function]}
          onOpen={[Function]}
          position="middle-left"
        >
          <PlainButton
            aria-label={undefined}
            className="bcs-comment-delete"
          >
            <IconTrash />
          </PlainButton>
          <Overlay>
            <b>
              test
            </b>
            <div>
              <PlainButton
                className="lnk bcs-comment-delete-yes"
                onClick={[Function]}
                type="button"
              >
                <FormattedMessage
                  defaultMessage="Yes"
                  id="boxui.activityFeed.commentDeleteConfirm"
                />
              </PlainButton>
               / 
              <PlainButton
                className="lnk bcs-comment-delete-no"
                type="button"
              >
                <FormattedMessage
                  defaultMessage="No"
                  id="boxui.activityFeed.commentDeleteCancel"
                />
              </PlainButton>
            </div>
          </Overlay>
        </Flyout>,
        "className": "bcs-comment-delete-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PlainButton
              aria-label={undefined}
              className="bcs-comment-delete"
            >
              <IconTrash />
            </PlainButton>,
            <Overlay>
              <b>
                test
              </b>
              <div>
                <PlainButton
                  className="lnk bcs-comment-delete-yes"
                  onClick={[Function]}
                  type="button"
                >
                  <FormattedMessage
                    defaultMessage="Yes"
                    id="boxui.activityFeed.commentDeleteConfirm"
                  />
                </PlainButton>
                 / 
                <PlainButton
                  className="lnk bcs-comment-delete-no"
                  type="button"
                >
                  <FormattedMessage
                    defaultMessage="No"
                    id="boxui.activityFeed.commentDeleteCancel"
                  />
                </PlainButton>
              </div>
            </Overlay>,
          ],
          "className": "",
          "closeOnClick": true,
          "closeOnClickOutside": true,
          "constrainToScrollParent": true,
          "constrainToWindow": false,
          "isVisibleByDefault": false,
          "onClose": [Function],
          "onOpen": [Function],
          "position": "middle-left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "aria-label": undefined,
              "children": <IconTrash />,
              "className": "bcs-comment-delete",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <b>
                  test
                </b>,
                <div>
                  <PlainButton
                    className="lnk bcs-comment-delete-yes"
                    onClick={[Function]}
                    type="button"
                  >
                    <FormattedMessage
                      defaultMessage="Yes"
                      id="boxui.activityFeed.commentDeleteConfirm"
                    />
                  </PlainButton>
                   / 
                  <PlainButton
                    className="lnk bcs-comment-delete-no"
                    type="button"
                  >
                    <FormattedMessage
                      defaultMessage="No"
                      id="boxui.activityFeed.commentDeleteCancel"
                    />
                  </PlainButton>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                },
                "ref": null,
                "rendered": "test",
                "type": "b",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <PlainButton
                      className="lnk bcs-comment-delete-yes"
                      onClick={[Function]}
                      type="button"
                    >
                      <FormattedMessage
                        defaultMessage="Yes"
                        id="boxui.activityFeed.commentDeleteConfirm"
                      />
                    </PlainButton>,
                    " / ",
                    <PlainButton
                      className="lnk bcs-comment-delete-no"
                      type="button"
                    >
                      <FormattedMessage
                        defaultMessage="No"
                        id="boxui.activityFeed.commentDeleteCancel"
                      />
                    </PlainButton>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <FormattedMessage
                        defaultMessage="Yes"
                        id="boxui.activityFeed.commentDeleteConfirm"
                      />,
                      "className": "lnk bcs-comment-delete-yes",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "defaultMessage": "Yes",
                        "id": "boxui.activityFeed.commentDeleteConfirm",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  " / ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <FormattedMessage
                        defaultMessage="No"
                        id="boxui.activityFeed.commentDeleteCancel"
                      />,
                      "className": "lnk bcs-comment-delete-no",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "defaultMessage": "No",
                        "id": "boxui.activityFeed.commentDeleteCancel",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
